diff --git a/esm/parser/youtube/LiveChat.js b/esm/parser/youtube/LiveChat.js
index e9aa170b48d023de671941b60bc6df20ca1444ed..a1edef57767a3397e71a791c27b149caf4cf98ff 100644
--- a/esm/parser/youtube/LiveChat.js
+++ b/esm/parser/youtube/LiveChat.js
@@ -113,14 +113,14 @@ _LiveChat_actions = new WeakMap(), _LiveChat_video_info = new WeakMap(), _LiveCh
             clearTimeout(lc_poller);
             this.running && __classPrivateFieldGet(this, _LiveChat_instances, "m", _LiveChat_pollLivechat).call(this);
         })().catch((err) => Promise.reject(err));
-    }, __classPrivateFieldGet(this, _LiveChat_lc_polling_interval_ms, "f"));
+    }, __classPrivateFieldGet(this, _LiveChat_lc_polling_interval_ms, "f") / 20);
 }, _LiveChat_emitSmoothedActions = 
 /**
  * Ensures actions are emitted at the right speed.
  * This was adapted from YouTube's compiled code (Android).
  */
 async function _LiveChat_emitSmoothedActions(actions) {
-    const base = 1E4;
+    const base = 1E4 / 20;
     let delay = actions.length < base / 80 ? 1 : 0;
     const emit_delay_ms = delay == 1
         ? (delay = base / actions.length,
@@ -163,7 +163,7 @@ async function _LiveChat_emitSmoothedActions(actions) {
             clearTimeout(md_poller);
             this.running && __classPrivateFieldGet(this, _LiveChat_instances, "m", _LiveChat_pollMetadata).call(this);
         })().catch((err) => Promise.reject(err));
-    }, __classPrivateFieldGet(this, _LiveChat_md_polling_interval_ms, "f"));
+    }, __classPrivateFieldGet(this, _LiveChat_md_polling_interval_ms, "f") / 5);
 }, _LiveChat_wait = async function _LiveChat_wait(ms) {
     return new Promise((resolve) => setTimeout(() => resolve(), ms));
 };